src/agent/communicator/src/http_client.cpp

```diff
@@ -282,10 +311,23 @@ namespace http_client
             const auto results = resolver->Resolve(params.Host, params.Port);
 
             auto socket = m_socketFactory->Create(io_context.get_executor(), params.Use_Https);
-            socket->Connect(results);
+            boost::beast::error_code ec;
+            socket->Connect(results, ec);
+
+            if (ec)
+            {
+                throw std::runtime_error("Error connecting to host: " + ec.message());
+            }
 
             const auto req = CreateHttpRequest(params);
-            socket->Write(req);
+
+            socket->Write(req, ec);
+
+            if (ec)
+            {
+                throw std::runtime_error("Error writing request: " + ec.message());
+            }
+
             socket->ReadToFile(res_parser, dstFilePath);
```

Comment:
No error code check here?

Reply:
The code of this function is pretty similar than `PerformHttpRequest`. Maybe we can create a private function for it?